storm-core/src/jvm/org/apache/storm/generated/Assignment.java

```diff
@@ -787,15 +787,15 @@ public void read(org.apache.thrift.protocol.TProtocol iprot, Assignment struct)
           case 2: // NODE_HOST
             if (schemeField.type == org.apache.thrift.protocol.TType.MAP) {
               {
-                org.apache.thrift.protocol.TMap _map548 = iprot.readMapBegin();
-                struct.node_host = new HashMap<String,String>(2*_map548.size);
-                String _key549;
-                String _val550;
-                for (int _i551 = 0; _i551 < _map548.size; ++_i551)
+                org.apache.thrift.protocol.TMap _map564 = iprot.readMapBegin();
```

Comment:
There's no dumb question. This is just a one of practices on Storm project. As always there're some pros and cons regarding this:

pros: Thrift compiler is only needed for contributors who modifies storm.thrift, and the case is actually not often.
cons: Thrift generates codes which makes huge diff even though just adding only one field.

IMO, installing thrift compiler because of just build project is bad, especially OSPs are using different versions of thrift. While disadvantage is also not a trivial, I think advantage is valuable enough.